tests/codex/testnode.nim

```diff
@@ -163,3 +195,112 @@ asyncchecksuite "Test Node":
     var data = newSeq[byte](testString.len)
     await stream.readExactly(addr data[0], data.len)
     check string.fromBytes(data) == testString
+
+
+asyncchecksuite "Test Node - host contracts":
+  let
+    (path, _, _) = instantiationInfo(-2, fullPaths = true) # get this file's name
+
+  var
+    file: File
+    chunker: Chunker
+    switch: Switch
+    wallet: WalletRef
+    network: BlockExcNetwork
+    clock: MockClock
+    localStore: RepoStore
+    localStoreRepoDs: DataStore
+    localStoreMetaDs: DataStore
+    engine: BlockExcEngine
+    store: NetworkStore
+    sales: Sales
+    node: CodexNodeRef
+    blockDiscovery: Discovery
+    peerStore: PeerCtxStore
+    pendingBlocks: PendingBlocksManager
+    discovery: DiscoveryEngine
+    manifest: Manifest
+    manifestCid: string
+
```

Comment:
This is a lot of duplication, can we do this in one place?

Reply:
Honestly, I don't really see a way, unless we want to run the above tests (the one in "Test Node" suite) with Host contracts enabled, it is an option, but I thought better to separate these.

Node object (and hence its tests) require a lot of dependencies and that is why there is lot of duplication. Also as tests depends on closure a lot, having some "setup utility function" is not really option here.